File: Practica 4/Ej3/potencia.c
#include <stdio.h>

int potencia(int base, int exponent)
{
    if (exponent == 0)
    {
        return 1;
    }

    int total = base;
    for (int i = 1; i < exponent; i++)
    {
        total *= base;
    }

    return total;
}

int main()
{
    int base, exponent;

    printf("Introduce un número entero positivo: ");
    scanf("%d", &base);

    while (base < 0)
    {
        printf("Error: El número debe ser positivo\n");

        printf("Vuelve a introducirlo: ");
        scanf("%d", &base);
    }

    printf("Introduce un número entero positivo: ");
    scanf("%d", &exponent);

    while (exponent < 0)
    {
        printf("Error: El número debe ser positivo\n");

        printf("Vuelve a introducirlo: ");
        scanf("%d", &exponent);
    }

    printf("%d elevado a %d es: %d", base, exponent, potencia(base, exponent));
}

File: Practica 6/Ej3/potencia.py
def main():
    n1 = int(input("Introduce un número entero positivo: "))
    n2 = int(input("Introduce otro número entero positivo: "))

    while n1 < 0 or n2 < 0:
        print("Error: Los números deben ser positivos")
        n1 = int(input("Introduce un número entero positivo: "))
        n2 = int(input("Introduce otro número entero positivo: "))

    print(f"{n1} elevado a {n2} es {potencia(n1, n2)}")


def potencia(b, e):
    if e == 0:
        return 1

    total = 1
    for n in range(e):
        total *= b

    return total


main()
